ts/features/payments/common/components/PaymentCard.tsx

```diff
@@ -0,0 +1,222 @@
+    <LabelSmallAlt
+      accessibilityLabel={I18n.t("wallet.methodDetails.a11y.bpay.owner", {
+        fullOwnerName: props.holderName
+      })}
+    >
+      {props.holderName}
+    </LabelSmallAlt>
+  );
+
+  const expireDateText = props.expireDate && (
+    <LabelSmallAlt>
+      {I18n.t("wallet.creditCard.validUntil", {
+        expDate: format(props.expireDate, "MM/YY")
+      })}
+    </LabelSmallAlt>
+  );
+
+  const maskedEmailText = props.holderEmail && (
+    <LabelSmallAlt
+      accessibilityLabel={I18n.t("wallet.methodDetails.a11y.paypal.owner", {
+        email: props.holderEmail
+      })}
+    >
+      {props.holderEmail}
+    </LabelSmallAlt>
+  );
+
+  const maskedPhoneText = props.holderPhone && (
+    <LabelSmallAlt
+      accessibilityLabel={I18n.t("wallet.methodDetails.a11y.bpay.phone", {
+        // we do this to make the screen reader read the number digit by digit,
+        phoneNumber: props.holderPhone.split("").join(" ")
+      })}
+    >
+      {props.holderPhone}
+    </LabelSmallAlt>
+  );
+
+  const renderBankLogo = () => {
+    if (props.holderEmail) {
+      return (
+        <View style={styleSheet.bankInfo}>
+          <PayPalLogo
+            testID="paymentCardPayPalLogoTestId"
+            accessible={true}
+            accessibilityLabel="PayPal"
+            height={32}
+            width={113}
+          />
+        </View>
+      );
+    }
+
+    if (props.holderPhone) {
+      return (
+        <View style={styleSheet.bankInfo}>
+          <BPayLogo
+            testID="paymentCardBPayLogoTestId"
+            accessible={true}
+            accessibilityLabel="BANCOMAT Pay"
+            height={24}
+            width={136}
+          />
+        </View>
+      );
+    }
+
+    if (props.abiCode) {
+      return (
+        <View style={styleSheet.bankInfo}>
+          <PaymentCardBankLogo
+            testID="paymentCardBankLogoTestId"
+            abiCode={props.abiCode}
+            height={24}
+          />
+        </View>
+      );
+    }
+
+    if (props.hpan) {
+      const circuitName =
+        props.brand || I18n.t("wallet.methodDetails.cardGeneric");
+
+      return (
+        <H6
+          accessibilityLabel={I18n.t("wallet.methodDetails.a11y.credit.hpan", {
+            circuit: circuitName,
+            // we space the hpan to make the screen reader read it digit by digit
+            spacedHpan: props.hpan.split("").join(" ")
+          })}
+          style={styleSheet.bankInfo}
+        >
+          {capitalize(circuitName)} •••• {props.hpan}
+        </H6>
+      );
+    }
+
+    return undefined;
+  };
+
+  return (
+    <View style={styleSheet.card}>
+      <View style={styleSheet.wrapper}>
+        <View style={styleSheet.paymentInfo}>
+          {renderBankLogo()}
+          {cardIcon}
+        </View>
+        <View style={styleSheet.additionalInfo}>
+          {holderNameText}
+          {expireDateText}
+          {maskedEmailText}
+          {maskedPhoneText}
```

Comment:
<img width="200" alt="Screenshot 2024-03-20 at 14 51 31" src="https://github.com/pagopa/io-app/assets/60693085/b980664d-fe91-4ace-8ceb-c93c8d4eaeef">

I've tried "filling all the fields" the component allows type-wise, and the result is as shown.

Now, I do understand that this is an edge case as well as a well handled one, but are you sure that you want to give this much freedom in development? 
It's just some thought, I overall like the implementation.

Reply:
Yes, it is intended. This makes the component totally agnostic towards the type of payment. It was a request made by us to the design team.
cc: @thisisjp